### ISSUE TITLE
[Bug] "Access Denied" on Akamai based websites

### ISSUE THREAD

**Issue by privacyguy123** [darkreader/darkreader#8774]
I see at least one other open ticket referencing this issue, but they seem to think it's a Minecraft.net specific issue and I think it goes deeper than that so I'm making my own more in depth ticket.

# Bug Report

## Website Address
Any website using Akamai CDN - we know this due to the "access denied" page it serves (example here is https://www.flannels.com)

## Bug Description
Dark Reader using Dynamic mode is editing something and Akamai scripts are picking that up as some sort of bot activity that is triggering a short 10 minute IP ban - this is my best guess, hopefully more minds here can chime in.

## Steps To Reproduce
Go to https://www.flannels.com (or https://www.cruisefashion.com) with Dark Reader on Dynamic mode. Click around until you get "banned" (access denied) - wait 10 minutes and repeat. 

## Expected Behavior
This obviously should not happen and is a pretty major flaw in an addon I rely on daily.

## Actual Behavior
Covered under Bug Description - I have found at least one other on the arkenfox/user.js Github thinking that it was perhaps a modified FireFox setting causing the IP ban to trigger.

## Screenshots
![image](https://user-images.githubusercontent.com/50337995/165402566-10f4e8e3-1e3d-40b0-849c-c8c604e86fe2.png)

## System Information
- OS: Windows 10 LTSC 2021
- Browser: FireFox 99.01
- Dark Reader Version: 4.9.48

**Comment by xcasxcursex** [comment]
I've been suffering this a while, just put it down to bad CDN and never tried to figure out what triggered it. Good work.

**Comment by privacyguy123** [comment]
Yeah that's the weird part, it's after 2 clicks deep for me - way over my head even beginning to understand it but it's not isolated to these 2 websites, it's a whole Akamai CDN thing. I guess it thinks that we are scraping data or DDoSing or something similar?

**Comment by xcasxcursex** [comment]
DR's theming is definitely local file inclusion and probably looks like XSS at a glance from a bot. This is an akamai false positive but good luck having them fix it. They will probably be exercising obscurity about it too, so working around a mystery from DR's end could be very difficult. 

All that you can really do is follow their advice on that page and use the page at https://www.akamai.com/us/en/clientrep-lookup/ to have them investigate. Ironically my browser blocked the captcha for tracking so I can't proceed past that page.

**Comment by Gusted** [comment]
> I think the ball is in DRs court to fix this

From dark reader perspective, it's required to make those requests. There is no other way of collecting the CSS that the site is using.

**Comment by privacyguy123** [comment]
Hopefully one of you lot know somebody with contacts within Akamai then!

**Comment by Gusted** [comment]
> Hopefully one of you lot know somebody with contacts within Akamai

Nothing on my end, but even then it's most likely "Seems like a you problem!"

**Comment by eygraber** [comment]
This happens to me constantly with delta.com

Extremely annoying on their end. Maybe there's someone in the community at large that has some sway there?

**Comment by JacksonWrath** [comment]
Seconded on Delta, and OpenTable's website is now also doing this to me. I've seen this on a number of other minor websites lately (most recently dyson.com).

**Comment by bardium** [comment]
Same happened to me on [https://www.minecraft.net](https://www.minecraft.net)

**Comment by Hellscap3d** [comment]
> Same happened to me on [minecraft.net](https://www.minecraft.net)

I second that

**Comment by danielzgtg** [comment]
Potentially related to #7235

**Comment by davewj100** [comment]
I have seen this as well. Another example is the Mellanox driver download page:

https://network.nvidia.com/products/adapter-software/ethernet/windows/winof-2/#tabs-2

When trying to download drivers, you get an access denied that looks exactly like the screenshot. Here is a direct link:

https://www.mellanox.com/downloads/WinOF/MLNX_WinOF2-3_10_51000_All_x64.exe

**Comment by csss1234** [comment]
Delta.com
westernunion.com

**Comment by danielzgtg** [comment]
Has anyone contacted Akamai?

**Comment by pallaswept** [comment]
Adding stan.com.au (Aussie netflix basically) to the list effected by this bug.

**Comment by pallaswept** [comment]
> Has anyone contacted Akamai?

You need to do it.

https://github.com/darkreader/darkreader/issues/8774#issuecomment-1116000742

**Comment by pallaswept** [comment]
> Now it appears that Cision/PR Newswire is broken! ![image](https://user-images.githubusercontent.com/71337477/258647378-5c089d3e-5469-4c9d-8e2a-c3637857e6f7.png) https://tripadvisor.mediaroom.com/2021-06-23-Tripadvisor-Reveals-New,-Expansive-Travelers-Choice-Things-to-Do-Awards-to-Help-Travelers-Maximize-Every-Trip-with-the-Best-Experiences



> It now appears that Dark Reader is breaking PRNewsWire. ![image](https://user-images.githubusercontent.com/71337477/258647464-eb99657b-606b-4cd7-b02b-becd210f968e.png) I disabled the extension and it works fine.



> > Has anyone contacted Akamai?
> 
> You need to do it.
> 
> > All that you can really do is follow their advice on that page and use the page at https://www.akamai.com/us/en/clientrep-lookup/ to have them investigate.
> > [Like this guy said](https://github.com/darkreader/darkreader/issues/8774#issuecomment-1116000742)

In my most recent report to Akamai, they changed their standard response from 'we'll look into it' and never responding again, and are now issuing a standardised response of 'tell the website operator, not us - the rate limiter is variable and is configured by each website using our services'

So, if you find a website that triggers the rate limiter, hit them up directly through their support pages, and let them know that an addon called Dark Reader with over 2 million users across every browser is unable to access their website because the rate limiter settings are too strict. The helpdesk operator you initially speak with, will be unlikely to understand this, so you may need to encourage them to escalate the issue to their web admin team. Pass on the link to this page and it'll hopefully reach the person in charge of fixing it up.

**Comment by sofatbobo** [comment]
Could Dark Reader be configured, or updated, to automatically detect Akamai and toggle off?

This was making me pull out my hair. So happy I found this thread. Delta.com is where I mainly encounter this.

**Comment by pallaswept** [comment]
> Could Dark Reader be configured, or updated, to automatically detect Akamai and toggle off?

No because the problem isn't akamai, it is individual websites' implementation of akamai's services being poorly configured, that is the problem. DR has the ability to be disabled on specific sites already, so you can do that until your report to that site is taken seriously and they fix their rate limiter settings.

**Comment by pallaswept** [comment]
Hilarious! Same applies, we have to disable DR on their site and report it to them so they fix it.

**Comment by pallaswept** [comment]
> Now it doesn't work with Subway! Access Denied You don't have permission to access "http://www.subway.com/en-US/" on this server. Reference #18.8a1c2117.1704475276.cbe08072

Can't reproduce that here. I can visit that site and click away all day and it never blocked me. If it is a DR thing it will effect us all, so this is something else. If disabling DR fixed it for you, it must be a combination of DR and something else on your system. Sorry bro. I hope you got your sandwich in the end. 

Screenshot to show it definitely works with DR enabled. I blacked out my tabs and addons for privacy in case you're wondering why it doesn't look like a real browser, I swear I didn't 'photoshop fake news' this :laughing: 

![image](https://github.com/darkreader/darkreader/assets/132128962/cbeab178-8eb4-48ff-a8c5-d59fc3de4bd2)

**Comment by mud-ali** [comment]
> Now it doesn't work with Subway! Access Denied You don't have permission to access "http://www.subway.com/en-US/" on this server. Reference #18.8a1c2117.1704475276.cbe08072

just tried this and the same thing happens for me. 
1. went to website with DR on, clicked a link and got access denied
2. globally disabled DR
3. went back to subway and clicked around a bunch, no access denied

i didnt change anything else about my system between that, just closed the tab and disabled dark reader

**Comment by pallaswept** [comment]
@mud-ali see the post above you.

**Comment by mud-ali** [comment]
@pallaswept yeah I saw that. The only thing that I changed was disabling dark reader though, so I don't exactly understand what "else on my system" it could be. Do you think it could be something to do with differing browsers? I'm on firefox 121.0 for arch linux, what about you?

![image](https://github.com/darkreader/darkreader/assets/96320211/4ce8b683-b51a-4784-b000-9753de2e5199)

**Comment by pallaswept** [comment]
> The only thing that I changed was disabling dark reader though, so I don't exactly understand what "else on my system" it could be. 

Yes but consider that the failure requires two parts simultaneously, DR+_mysterything_. You disable DR, now you don't have the two parts simultaneously, and you don't get the failure.
Me, I don't have _mysterything_. So I have DR enabled, but I also don't have the two parts simultaneously, and I don't get the failure....

The _mysterything_ could be so many different things, I could never imagine. Good thinking to check something simple first. I'm on FF 121 on OpenSuSE Tumbleweed. I guess if you had the time to spare, and you have a VM host setup, you could download the Tumbleweed live iso and boot it in a VM, install DR, and visit subway, and see if it works. If that also fails, it's likely that _mysterything_ is something related to your network (as in, your ISP, not your LAN)

**Comment by pallaswept** [comment]
> And another one! Weather Underground! Looks like no DR for my weather station! 😠 Service Unavailable - DNS failure The server is temporarily unable to service your request. Please try again later. Reference #11.102b3417.1704676143.cbbf3e6c ![image](https://private-user-images.githubusercontent.com/71337477/294796305-1a943c8a-5f44-4ff1-bb2e-898332b3616c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ2ODQxMTgsIm5iZiI6MTcwNDY4MzgxOCwicGF0aCI6Ii83MTMzNzQ3Ny8yOTQ3OTYzMDUtMWE5NDNjOGEtNWY0NC00ZmYxLWJiMmUtODk4MzMyYjM2MTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDAzMTY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxYjVjOWZlMTBhMTYzNDJiZmUyMjMyMjUxNTdhODU2ODM3MTc1OWE3NTBkNDdkZGQ5MWRkZDBhOTY1MzAzYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pMT_qiZz5ioRG5SmA2rpkAjVAE2Up26t4nBosZ43rxQ)

Do you mean wunderground.com? That's also working for me with DR enabled.

**Comment by FerbKB** [comment]
Another data point here but wunderground works fine for me, but gamestop and subway didn't until I disabled DR. I was able to replicate this with machines outside of my network, so it's not ISP or LAN related.

**Comment by pallaswept** [comment]
> outside of my network, so it's not ISP or LAN related.

Akamai can discriminate on all manner of points all the way from you to the target site, so it might not even be something inside your country, but still be enough to get them to block you....

Gamestop didn't work here, either, so yeh that's another one to add to the list of sites that need to configure their Akamai implementation.

Subway definitely works with DR, though, see the screenshot above (and I just tested it again to make sure, and I tried local subway (I'm in AU) and it's still working):
https://github.com/darkreader/darkreader/issues/8774#issuecomment-1879666239

Just to accentuate the point that akamai can discriminate on all manner of points along the path from you to them, I thought I'd try the subway site with dark reader, using a VPN with an endpoint in the US, and guess what? Blocked. So that explains why I can get there but you guys can't. Seems like they're only rate-limiting on US sources.

As always, all we can do is disable dark reader and tell them they have their site mis-configured and are consequently blocking millions of users.

**Comment by pallaswept** [comment]
Another site for the list. A big one this time:  https://www.nvidia.com/


Log of the chat with Nvidia support FYI. Hopefully they'll sort that one out pretty soon.

Hi, my name is (name deleted for their privacy). How may I help you?
Hello

Hi (name). I hope you are well today
I have a problem which I am sure you will be unable to fix, but I hope will be able to escalate this issue to the appropriate team for them to rectify it

Please be assured that I will do my best to help you further or point you in the right direction.
Could you please elaborate the concern?

First, to explain the consequence: millions of users will be unable to access nvidia.com
Now, the cause....
nvidia.com is being hosted by akamai. They provide a function to prevent abuse of their customers' (such as nvidia) websites, a 'rate limiter', which prevents clients from making too many requests
in a short amount of time. This is a good thing, as it prevents DDOS attacks from bringing down the website
HOWEVER

Okay

The rate limiter can be configured, to set the degree to which it will allow client requests
if it is too tight, the rate limiter can prevent legitimate users from accessing the site
There is a browser addon, called 'Dark Reader' - you may know of it, it is one of the most popular browser extensions in every browser.
the rate limiter has been set, such that the requests made by Dark Reader during normal usage, trigger the rate limiter to block the client's IP address as though they were an attacker
The solution to this, is to have the website administrator reconfigure their akamai implementation to relax the restriction applied by the rate limiter, such that legitimate users can still access the
website, while still being sufficient to prevent denial of service attacks to the website

Thank you for elaborating the issue.

This is a known issue with akamai hosting being incorrectly configured. If you would like, I can send you a link to the github issue log for the addon whee further discussion is made. 

I appreciate your effort in bring this to our attention. We will definitely look into this further.

I would also request you to provide a feedback from your end so that our developer team can look into it. Please see this page for the steps : https://nvidia.custhelp.com/app/answers/detail/a_id/4681

OK sure I will do so.
I'm afraid that process is not one I can follow, as I do not use an of those devices.
any of*
I am on a PC (linux) with a 3090 and 1070
Is there some other steps you'd like me to take, or do you think perhaps you might be able to escalate with the information I have provided?

No, that's alright.
I will let the backend team know this concern.
Is there anything else I can help you with on this chat?

Thank you very much
That's all for today. I hope you have a good weekend.

Thanks again :)

**Comment by pallaswept** [comment]
> @pallaswept Cant reproduce the issue. Nvidia is showing up ok.

And for me, too! It worked as recently as a couple of days ago, and today, checking driver versions, no joy. It was _definitely_ DR doing it, I tested on two browsers and two machines from three networks.... all failed, and worked 10 minutes later with DR disabled, then failed immediately with DR enabled again.

I guess they already fixed it! 

### Bravo, nvidia!!
Showing all these other websites how quickly it CAN be fixed.

Thanks for letting me know, @brmonaghan :smiley: I would have never bothered to check so soon, if you hadn't mentioned it.

**Comment by Lycerion** [comment]
Are you telling me, that I can get an entire airport network temporarily banned from accessing Delta.com, just by accidentally trying to load it in dark mode?

And that this has been an issue for years?

**Comment by pallaswept** [comment]
> Are you telling me, that I can get an entire airport network temporarily banned from accessing Delta.com, just by accidentally trying to load it in dark mode?
> 
> And that this has been an issue for years?

Temporarily? if I lived stateside I'd make a $10 wifi device that goes into a random wall socket somewhere subtle, and leave one constantly retrying in every airport :D
Do it right after betting against delta's stock prices and make billions.

Maybe they'd start to act like they cared. I wouldn't bet on it though.

### PS this is a joke and surely super illegal
 (although our legal systems are so far behind on tech and so lax on financial criminals that I wouldn't bet on that, either), 
### don't do this. It's just a joke

**Comment by pallaswept** [comment]
> Are you telling me, that I can get an entire airport network temporarily banned from accessing Delta.com, just by accidentally trying to load it in dark mode?
> 
> And that this has been an issue for years?

Yes. Or, not temporarily, or accidentally.

I made another post as a joke, mentioning how it could be used maliciously, and it's so dangerous I decided to delete it before I got SWATed. But Delta don't even care. It's baffling and just reeks of incompetence.

**Comment by Proaxel** [comment]
This doesn't contribute to solution, but rather is for anyone googling around this problem like I was:
 do suspect that this also occurs on American Airlines website, particularly newjetnet.aa.com:
![image](https://github.com/darkreader/darkreader/assets/9545347/1938cbbf-879e-4924-8ee3-ab5563d22f0a)

The ban happens as soon as I log in. If I wait some time for the ban to expire and try again on either a fresh alternate browser (Edge) or an incognito window with all extensions disabled, the ban doesn't occur.

**Comment by cibere** [comment]
I'm getting this error with `answers.microsoft.com` and `sparknotes.com`
Though with sparknotes, it only happens when going to this specific URL: `https://www.sparknotes.com/lit/christmascarol/section1/`.

Before I figured out that darkreader was the problem, I did some digging, and If I understand what I found correctly, darkreader is causing [UCEPROTECT](dnsbl-3.uceprotect.net) to flag my IP as spam and temporarily blacklist my IP.

**Comment by cibere** [comment]
> > Perhaps it's a problem with the settings I have on the other profile?
> 
> That's a very interesting observation. I wonder if perhaps you might be able to export the settings, so they could be compared, and tested? Perhaps the DR devs could use that info.

I suppose I could, though I just tested it again to make sure, and I still get the error even with dark reader being disabled, however when I posted my original comment, it only got blocked when darkreader was enabled. (and I'm talking about the same profile, not the one I'm using with github)

[The one where I get the error](https://github.com/user-attachments/files/17551229/Not_working_dark_reader_settings.json)
[The one where I don't get the error](https://github.com/user-attachments/files/17551230/Working_Dark_Reader_Settings.json)

**Comment by privacyguy123** [comment]
Hey guys - reporting back in 2 years later and it's still an issue. Now with a much bigger website ... Royal Mail! I cannot even track my post now because they are using Akamai CDNs still using this heuristic style blocking.

**Comment by alexanderby** [comment]
I wonder if you try Firefox version 4.9.97. It should significantly reduce the number of requests to background images.

**Comment by stormsnsnow** [comment]
Broken again, this time with Dell
Access Denied
You don't have permission to access "http://www.dell.com/di/trustelevation/mfa_post" on this server.
Reference #18.570c0317.1758215216.549443c4

https://errors.edgesuite.net/18.570c0317.1758215216.549443c4

**Comment by stormsnsnow** [comment]
Another one is broken
This time with FLETC in NM. Possibly going there as part of training for new job.
Access Denied
You don't have permission to access "http://edit.fletc.gov/sites/default/files/FLETC%20Student%20computer%20requirements%2020220217_0.pdf" on this server.
Reference #18.250f3417.1767632278.3b5752ae

https://errors.edgesuite.net/18.250f3417.1767632278.3b5752ae

**Comment by alexanderby** [comment]
@stormsnsnow It gets blocked even if Dark Reader is not installed.

**Comment by stormsnsnow** [comment]
Another one, this time weather underground
Service Unavailable - DNS failure
The server is temporarily unable to service your request. Please try again later.
Reference #11.1a643017.1768075056.d84e61ee

https://errors.edgesuite.net/11.1a643017.1768075056.d84e61ee

**Comment by danielzgtg** [comment]
> DNS failure

@stormsnsnow Dark Reader has nothing to do with DNS.

**Comment by stormsnsnow** [comment]
Another one, this time Outback
Access Denied
You don't have permission to access "http://www.outback.com/order" on this server.
Reference #18.1713d517.1769806947.2fde282f

https://errors.edgesuite.net/18.1713d517.1769806947.2fde282f

**Comment by stormsnsnow** [comment]
I was however able to place said order with Edge. Works fine there.

**Comment by pallaswept** [comment]
I think we all understand that this issue exists and we don't need to be pinged every time a new example is found.

**Comment by stormsnsnow** [comment]
Another one! Access Denied
You don't have permission to access "http://www.choicehotels.com/" on this server.

Reference #0.4a0f3417.1775900687.6b59df19

https://errors.edgesuite.net/0.4a0f3417.1775900687.6b59df19
